overlord/snapstate/handlers.go

```diff
@@ -877,6 +877,23 @@ func (m *SnapManager) doSwitchSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *SnapManager) doToggleSnapFlags(t *state.Task, _ *tomb.Tomb) error {
```

Comment:
The name `toggle` seems to imply we flip some value to the opposite state. I found this surprising as the code simply assigns snap state from snap setup. Unless I'm missing some context perhaps the name could be `do{Transfer,Assign,Set}SnapFlags`?

Reply:
I called it this way because is analogous to switch-snap-channel, which is not called set-snap-channel (though is also sort of what it does), this would never be called if there is no flag to flip.

I can see both sides though, don't have a strong opinion